File: javascript/collatz-conjecture/collatz-conjecture.js
export const steps = (num) => {
  if(num < 1) {
    throw new Error('Only positive numbers are allowed')
  }

  for(var steps = 0; 1 < num; steps++) {
    num = 
      num % 2 === 0 ? num/2 : num * 3 + 1;
  }
  
  return steps
};

File: ruby/collatz-conjecture/collatz_conjecture.rb
class CollatzConjecture
    def self.steps(num)
        raise ArgumentError if num < 1

        steps = 0
        while num > 1
            num = num % 2 === 0 ? num / 2 : num * 3 + 1
            steps += 1
        end
        steps
    end
end